src/spartan/polynomial.rs

```diff
@@ -46,6 +61,22 @@ impl<Scalar: PrimeField> EqPolynomial<Scalar> {
   }
 }
 
+/// A multilinear extension of a polynomial $Z(\cdot)$, donate it as $\tilde{Z}(x_1, ..., x_m)$
```

Comment:
This is just a multilinear polynomial, not necessarily a multilinear extension, right? Multilinear polynomials can stand on their own, and they don't need to necessarily be extensions of other polynomials.

Reply:
Yeah, this comment came from https://github.com/privacy-scaling-explorations/Nova/pull/16 - can update it a bit to be more accurate

Reply:
I think in this contex, we should just update the name to `MLEPolynomial` or similar. 

As we never define multilinear polynomials in the lib unless they're MLEs right??